### PR TITLE
Request history UI

### PR DESCRIPTION
**Pull request by PrabSG**
RequestCompletedUI
RequestPendingUI

### CODE REVIEW

File: coffee-chats/src/main/webapp/src/entity/Requests.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  matchRandom: boolean,
+  matchRecents: boolean,
+  userId: string,
+  requestId: number
+};
+
+export type CompletedRequest = ExpiredRequest | MatchedRequest;
+
+export interface MatchedRequest {
+  matched: boolean,
```
Comment (TsarN) [inline coffee-chats/src/main/webapp/src/entity/Requests.ts:16]: A lot of repeated code, maybe create an interface with common fields and inherit from it?

File: coffee-chats/src/main/webapp/src/util/chatRequest.ts [hunk 2]
```diff
@@ -49,3 +51,44 @@ export const submitChatRequest = async (interests: string[],
...
 };
+
+/**
+ * Fetches pending requests from backend for the current user, updates state with the response via 
+ * given function.
+ * 
+ * @param setPendingRequests - Function to set the current pending requests user has.
+ */
+export const fetchPendingRequests = (setPendingRequests: (requests: ChatRequest[]) => void) => {
+  fetch(getRequestsURL('pending'))
```
Comment (TsarN) [inline coffee-chats/src/main/webapp/src/util/chatRequest.ts:62]: Why not just use async functions?

Comment (PrabSG) [reply coffee-chats/src/main/webapp/src/util/chatRequest.ts:62]: They are pretty much identical in behaviour, it just looked clearer here using fetch and then to show the data transformation from response to actual required objects.

File: coffee-chats/src/main/webapp/src/entity/Requests.ts (hunk 1 above)
Comment (PrabSG) [reply coffee-chats/src/main/webapp/src/entity/Requests.ts:20]: I opted to not do this initially for clarity and separation, but the majority of two of the request types are duplicated. I have refactored this now.

**Review by med0b1011 — APPROVED** [review]

**Review by TsarN — APPROVED** [review]

File: coffee-chats/src/main/webapp/src/App.tsx [hunk 3]
```diff
@@ -10,6 +10,7 @@ import {GroupListPage} from "./pages/GroupListPage";
 import {GroupInfoPage} from "./pages/GroupInfoPage";
 import {GroupEditPage} from "./pages/GroupEditPage";
 import {OAuthDialog} from "./components/OAuthDialog";
+import { RequestsPage } from "./pages/RequestsPage";
```
Comment (DarthGandalf) [inline coffee-chats/src/main/webapp/src/App.tsx:13]: inconsistent spaces

File: coffee-chats/src/main/webapp/src/components/MatchedRequestDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
+import React, { useState, useEffect } from 'react'
+import {MatchedRequest} from '../entity/Requests'
+import {
+  Grid, Box, Divider, Chip, makeStyles, Theme, createStyles, ListItemAvatar, List, ListItem, 
+  Avatar, ListItemText
+} from '@material-ui/core'
+import {format} from 'date-fns'
```
Comment (DarthGandalf) [inline coffee-chats/src/main/webapp/src/components/MatchedRequestDetails.tsx:7]: inconsistent spaces

File: coffee-chats/src/main/webapp/src/entity/Requests.ts [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  dateRanges: DateRangeResponse[],
+  minPeople: number,
+  maxPeople: number,
+  duration: DurationResponse,
+  matchRandom: boolean,
+  matchRecents: boolean,
+  userId: string
+}
+
+export interface ChatRequestResponse extends RequestParamsResponse{
```
Comment (DarthGandalf) [inline coffee-chats/src/main/webapp/src/entity/Requests.ts:55]: missing space

why you don't tell automation to format code for you?

File: coffee-chats/src/test/java/com/google/step/coffee/entity/ChatRequestBuilderTest.java [hunk 6]
```diff
@@ -70,4 +75,13 @@ public void mustHaveValidChatDuration() {
     }
   }
 
+  @Test
```
Comment (DarthGandalf) [inline coffee-chats/src/test/java/com/google/step/coffee/entity/ChatRequestBuilderTest.java:78]: All that code required only a single test?

Comment (PrabSG) [reply coffee-chats/src/test/java/com/google/step/coffee/entity/ChatRequestBuilderTest.java:78]: The only backend code I wrote was fetching pre-existing entities, so I am not sure what to test other than the small behavioural change I made to ChatRequestBuilder, which I did so here. The large amount of code is largely front end component code to display the fetched data in a visually pleasing format.

File: coffee-chats/src/main/webapp/src/entity/Requests.ts (hunk 5 above)
Comment (PrabSG) [reply coffee-chats/src/main/webapp/src/entity/Requests.ts]: I previously had done so, however it was not working well with the react TSX elements and breaking the spacing within those. Perhaps with more time I would configure this properly.

**Pull request merged**